codersdk/workspaceagents.go

```diff
@@ -144,42 +144,56 @@ const (
+	Name                 string                  `json:"name"`
+	ResourceID           uuid.UUID               `json:"resource_id" format:"uuid"`
+	InstanceID           string                  `json:"instance_id,omitempty"`
+	Architecture         string                  `json:"architecture"`
+	EnvironmentVariables map[string]string       `json:"environment_variables"`
+	OperatingSystem      string                  `json:"operating_system"`
+	LogsLength           int32                   `json:"logs_length"`
+	LogsOverflowed       bool                    `json:"logs_overflowed"`
+	Directory            string                  `json:"directory,omitempty"`
+	ExpandedDirectory    string                  `json:"expanded_directory,omitempty"`
+	Version              string                  `json:"version"`
+	Apps                 []WorkspaceApp          `json:"apps"`
 	// DERPLatency is mapped by region name (e.g. "New York City", "Seattle").
-	DERPLatency              map[string]DERPRegion `json:"latency,omitempty"`
-	ConnectionTimeoutSeconds int32                 `json:"connection_timeout_seconds"`
-	TroubleshootingURL       string                `json:"troubleshooting_url"`
-	// Deprecated: Use StartupScriptBehavior instead.
-	LoginBeforeReady             bool                 `json:"login_before_ready"`
-	ShutdownScript               string               `json:"shutdown_script,omitempty"`
-	ShutdownScriptTimeoutSeconds int32                `json:"shutdown_script_timeout_seconds"`
-	Subsystems                   []AgentSubsystem     `json:"subsystems"`
-	Health                       WorkspaceAgentHealth `json:"health"` // Health reports the health of the agent.
-	DisplayApps                  []DisplayApp         `json:"display_apps"`
+	DERPLatency              map[string]DERPRegion     `json:"latency,omitempty"`
+	ConnectionTimeoutSeconds int32                     `json:"connection_timeout_seconds"`
+	TroubleshootingURL       string                    `json:"troubleshooting_url"`
+	Subsystems               []AgentSubsystem          `json:"subsystems"`
+	Health                   WorkspaceAgentHealth      `json:"health"` // Health reports the health of the agent.
+	DisplayApps              []DisplayApp              `json:"display_apps"`
+	LogSources               []WorkspaceAgentLogSource `json:"log_sources"`
+	Scripts                  []WorkspaceAgentScript    `json:"scripts"`
+}
+
+type WorkspaceAgentLogSource struct {
+	WorkspaceAgentID uuid.UUID `json:"workspace_agent_id" format:"uuid"`
+	ID               uuid.UUID `json:"id" format:"uuid"`
+	CreatedAt        time.Time `json:"created_at" format:"date-time"`
+	DisplayName      string    `json:"display_name"`
+	Icon             string    `json:"icon"`
+}
+
+type WorkspaceAgentScript struct {
+	LogSourceID      uuid.UUID     `json:"log_source_id" format:"uuid"`
+	LogPath          string        `json:"log_path"`
+	Source           string        `json:"source"`
+	CRON             string        `json:"cron"`
```

Comment:
I haven't seen cron stylized in upper case before, this feels more natural:

```suggestion
	Cron             string        `json:"cron"`
```